src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -60,80 +59,93 @@ const VideoInputBackgroundBlurControl: React.FC<Props> = ({
   useEffect(() => {
     meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
     return () => {
-      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+      meetingManager.unsubscribeFromSelectedVideoInputTransformDevice(
+        setDevice
+      );
     };
   }, []);
 
-  async function toggleBackgroundBlur() {
+  const toggleBackgroundBlur = async (): Promise<void> => {
     let current = activeVideoDevice;
-    if (isLoading) {
+    if (isLoading || !current) {
       return;
     }
     setIsLoading(true);
-    if (!isVideoTransformDevice(current)) {
-      // Enable video transform on the non-transformed device
-      current = await createBackgroundBlurDevice(current);
-      meetingManager.logger?.info(
-        'Video filter turned on - selecting video transform device: ' +
-          JSON.stringify(current)
-      );
-    } else {
-      // switch back to the inner device
-      current = await current.intrinsicDevice();
-      meetingManager.logger?.info(
-        'Video filter was turned off - selecting inner device: ' +
-          JSON.stringify(current)
-      );
+
+    try {
+      if (!isVideoTransformDevice(current)) {
+        // Enable video transform on the non-transformed device
+        current = await createBackgroundBlurDevice(current);
+        meetingManager.logger?.info(
+          'Video filter turned on - selecting video transform device: ' +
+            JSON.stringify(current)
+        );
+      } else {
+        // switch back to the inner device
+        current = await current.intrinsicDevice();
+        meetingManager.logger?.info(
+          'Video filter was turned off - selecting inner device: ' +
+            JSON.stringify(current)
+        );
+      }
+      // If we're currently using a video transform device, and a non-video transform device is selected
+      // then the video transform device will be stopped automatically
+      await selectVideoinput(current);
+    } catch (error) {
+      console.error('Failed to toggle Background Blur');
     }
-    // If we're currently using a video transform device, and a non-video transform device is selected
-    // then the video transform device will be stopped automatically
-    await meetingManager.selectVideoInputDevice(current);
+
     setIsLoading(false);
-  }
+  };
+
+  const handleClick = async (deviceId: string): Promise<void> => {
+    try {
+      // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
+      // If background blur is not on, then do a normal video selection
+      if (isVideoTransformDevice(activeVideoDevice) && !isLoading) {
+        setIsLoading(true);
+        if ('chooseNewInnerDevice' in activeVideoDevice) {
```

Comment:
Don't think we need to check for it

Reply:
So `chooseNewInnerDevice` is only available for `VoiceFocusTransformDevice`. I don't know why they check this for a video input device.
I think this is wrong here. and this if condition can never be passed.

Reply:
`chooseNewInnerDevice` is only available in `DefaultVideoTransformDevice`. This is to check if the `activeVideoDevice` is a `DefaultVideoTransformDevice`. So it's needed.

Reply:
Man this should just be in the interface :)

Reply:
Can you create a SIM to track this in V4